ledger/src/blockstore_db.rs

```diff
@@ -742,7 +742,10 @@ impl Rocks {
     }
 
     fn write(&self, batch: RWriteBatch) -> Result<()> {
-        let is_perf_context_enabled = maybe_collect_perf_context(self.3.rocks_perf_sample_interval);
+        let is_perf_context_enabled = maybe_collect_perf_context(
```

Comment:
`maybe_enable_perf_context` might be a better name.  The function not only returns the decision but also enables the PerfContext.  So the code looks like this:

```rust
  let is_perf_context_enabled = maybe_collect_perf_context(...); // this may enable PerfContext
  some_rocksdb_operation;  // if the perf context is enabled above, then RocksDB will record perf metrics under PerfContext
  if s_perf_context_enabled {
    ...  // we report the collected metrics (or later on we update the aggregated struct here).
  }

Reply:
I don't want to block you on this; so whatever you want to name it here, it is fine with me;
However just fyi and something to consider for future changes:

* As someone reading this code, I indeed found this name confusing and a speed bump.
* A function name does not necessarily need to expose its internal implementation details:
  * That only tends to add noise and distraction from the logic that the higher lever call stack needs to focus on.
  * The intention of the function is to abstract out some implementation and just expose the bare minimum that the call-site needs to worry about.
* In this case here, I don't think the call-site needs to care or even know what `perf_context` is; All it needs to know is if it "should collect perf" and if so then call this other method to report metrics. As such I find the "perf_context" in the name superfluous/noise.

But again your call. Please don't feel blocked on this.

Reply:
You're right about this.  It's better for this function name not tied to how it's implemented.  My previous comment is more about the function prefix where "should_" functions give me the impression that the function only does the boolean check and does not actually do any action while "maybe_" implies that the function may actually do something.

Reply:
> "maybe_" implies that the function may actually do something.

My point is in this case, the call-site does not care what the function does, or if does anything at all.
That information is only a distraction and noise if it is leaked to higher levels of the call stack.